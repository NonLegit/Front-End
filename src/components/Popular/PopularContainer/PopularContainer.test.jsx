import renderer from 'react-test-renderer';
import { ThemeProvider } from '@mui/system';
import { MemoryRouter as Router } from 'react-router-dom';
import PopularContainer from './PopularContainer';
import theme from '../../../styles/theme';
import ListingContextProvider from '../../../contexts/ListingContext';
import PostTypeContextProvider from '../../../contexts/PostTypeContext';
import HiddenPostsContextProvider from '../../../contexts/HiddenPostsContext';
import EditPostContextProvider from '../../../contexts/EditPostContext';
import CreatePostSidebarContextProvider from '../../../contexts/CreatePostSidebarContext';

describe(PopularContainer, () => {
  it('PopularContainer renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <CreatePostSidebarContextProvider>
            <ListingContextProvider>
              <PostTypeContextProvider>
                <HiddenPostsContextProvider>
                  <EditPostContextProvider>
                    <ThemeProvider theme={theme}>
                      <PopularContainer />
                    </ThemeProvider>
                  </EditPostContextProvider>
                </HiddenPostsContextProvider>
              </PostTypeContextProvider>
            </ListingContextProvider>
          </CreatePostSidebarContextProvider>
          ,
        </Router>,

      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
