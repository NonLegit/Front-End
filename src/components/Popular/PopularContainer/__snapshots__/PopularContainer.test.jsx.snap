// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularContainer PopularContainer renders correctly 1`] = `
Array [
  <div
    className="MuiBox-root css-pwymh4"
  >
    <div
      className="css-1yelhn"
    >
      <div
        className="css-3ggxsn"
        width={640}
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-9ef0pa-MuiTypography-root"
        >
          Popular posts
        </p>
        <div
          className="css-1i4l6gc"
        />
      </div>
    </div>
  </div>,
  ",",
]
`;
